File: 1.两数之和.py
# @before-stub-for-debug-begin
from python3problem1 import *
from typing import *
# @before-stub-for-debug-end


#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
	def twoSum(self, nums: List[int], target: int) -> List[int]:
		return_index = []

		for index, i in enumerate(nums):
			for j in nums[index+1:]:
				if i +j == target: 
					return_index.append(nums.index(i))
					nums.remove(i)
					return_index.append(nums.index(j) + 1)
		return return_index
		
# @lc code=end


# #
# @lc app=leetcode.cn id=1 lang=python
#
# [1] 两数之和
#

File: 1.两数之和.cpp
#include <unordered_map>
#include <vector>

using namespace std;

/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
public:
	vector<int> twoSum(vector<int>& nums, int target)
	{
		// 存储数字和其索引
		std::unordered_map<int, int> num_map;

		for (int i = 0; i < nums.size(); ++i) {
			int current_num = nums[i];
			int complement = target - current_num;

			// 检查补数是否存在于哈希表中
			if (num_map.count(complement)) {
				// 如果找到，返回补数和当前数字的索引
				return { num_map[complement], i };
			}

			// 如果没有找到补数，将当前数字和其索引添加到哈希表中
			num_map[current_num] = i;
		}

		// 根据题目描述，解总是存在的，
		// 所以这部分理论上不应该被执行到。
		// 如果不保证有解，返回空向量或抛出异常。
		return {};
	}
};
// @lc code=end
